app/src/main/java/cl/freedom/wheatermvvm/data/repository/CovidRepository.kt

```diff
@@ -1,14 +1,59 @@
 package cl.freedom.wheatermvvm.data.repository
+import androidx.lifecycle.LiveData
 import cl.freedom.desafiomvvm.data.network.MyApi
+import cl.freedom.wheatermvvm.data.dao.CurrentCovidDao
+import cl.freedom.wheatermvvm.data.entity.CurrentCovidEntryMapper
 import cl.freedom.wheatermvvm.data.network.SafeApiRequest
+import cl.freedom.wheatermvvm.data.response.CovidNetworkDataSource
 import cl.freedom.wheatermvvm.data.response.CovidResponse
-import retrofit2.Response
-import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class CovidRepository(val api : MyApi) : SafeApiRequest()
+class CovidRepository(private val currentCovidDao: CurrentCovidDao,
+                      private val covidNetworkDataSource: CovidNetworkDataSource)
 {
-    suspend fun getData(date : String? = null) : CovidResponse?
+    init {
+        covidNetworkDataSource.downloadedCurrentCovid.observeForever{
+            newCurrentCovid ->
+            persistFetchedCurrentCovid(newCurrentCovid)
+        }
+    }
+
+    private suspend fun initCovidData(date : String)
+    {
+        if(isFetchedCurrentNeeded(date))
+        {
+            fetchCurrentCovid(date)
+        }
+    }
+
+    private suspend fun fetchCurrentCovid(date : String)
+    {
+        covidNetworkDataSource.fetchCurrentCovid(date)
+    }
+
+
+    suspend fun getCurrentCovid(date : String) : LiveData<CurrentCovidEntryMapper>
     {
```

Comment:
you can join all this function in one... try to use the liveDataScope

Reply:
ok! I will Implement it!